client/jetpack-cloud/index.js

```diff
@@ -52,8 +55,31 @@ const redirectToPrimarySiteLanding = ( context, next ) => {
 
 const landingController = ( context, next ) => {
 	debug( 'controller: landingController', context );
-	context.primary = <Landing />;
-	next();
+
+	const state = context.store.getState();
+	const selectedSiteId = getSelectedSiteId( state );
+
+	// This path requires a selected site ID;
+	// if we don't have one, go to the main landing page
+	if ( ! Number.isFinite( selectedSiteId ) ) {
```

Comment:
Not a review: I hadn't seen isFinite before.  Are there situations where it's preferable to looking for a null instead?

Reply:
> Are there situations where it's preferable to looking for a null instead?

_Hypothetically_, if you're trying to determine whether a number input is a "normal" number, `Number.isFinite` is the function to use. In this particular case, I don't think it makes much of a difference other than to signal we're looking at a variable that should contain a numeric value of some kind. The possibilities shape up like so (notice how the outputs for `0`, `Infinity`, and `'abc'` change):

`Number.isFinite`:

| Input     | Output |
| --------- | ------ |
| 4         | true   |
| 0         | true   |
| NaN       | false  |
| Infinity  | false  |
| null      | false  |
| undefined | false  |
| 'abc'     | false  |

Truthy check (e.g., `if ( x )`):

| Input     | Output |
| --------- | ------ |
| 4         | true   |
| 0         | false  |
| NaN       | false  |
| Infinity  | true   |
| null      | false  |
| undefined | false  |
| 'abc'     | true   |